### DISCUSSION TITLE
Service Provider Registeration

### DISCUSSION THREAD

**Discussion by hamzaelmaghari** [laravel/framework#44889] [Ideas]
Hi, devs 👋
I have a feature request which is register `ServiceProvider` automaticaly once it's generated by the command (flag) `php artisan make:provider ExampleServiceProvider --register` once it's generated this latter should be registered in the `config/app.php`file automatically.

**Comment by ousid** [comment]
It seems a good idea. How about creating the service provider manually?

**Reply by hamzaelmaghari** [reply]
Laravel is a command line based framework, why caring about creating providers manualy?
There is no problem on creating providers manualy as well as you registering them in the `config/app.php` file.

**Reply by ousid** [reply]
I don't think, I agree with you on 
>Laravel is a command line based framework

And I can't disagree either, although, It's just a question, because I created a project outside the comfort of the base structure of Laravel, and I faced some difficulties using the artisan command when generating files.

I think making a PR to the framework is better, to find out whether is a good idea or not.

Good luck.

**Reply by henzeb** [reply]
I'm not a big fan of having those generator tools in the framework. They are more a plugin worthy type of functionality of the IDE. like Laravel IDE. https://plugins.jetbrains.com/plugin/13441-laravel-idea. With this I don't need to leave the IDE.
